src/migrations.zig

```diff
@@ -21,26 +21,33 @@ pub fn registerBlockMigrations(migrations: *std.StringHashMap(ZonElement)) void
 fn register(
 	collection: *std.StringHashMap([]const u8),
 	assetType: []const u8,
-	name: []const u8,
+	addonName: []const u8,
 	migrationZon: ZonElement,
 ) void {
+	const localAllocator = main.stackAllocator;
+
 	if((migrationZon != .object or migrationZon.object.count() == 0)) {
 		return;
 	}
 
 	var migrationZonIterator = migrationZon.object.iterator();
 	while(migrationZonIterator.next()) |migration| {
-		const result = collection.getOrPut(migration.key_ptr.*) catch unreachable;
+		const assetId = std.fmt.allocPrint(localAllocator.allocator, "{s}:{s}", .{addonName, migration.key_ptr.*}) catch unreachable;
```

Comment:
Why not just put in the arena allocator, and leak it if there is an error, it's really not that big of a deal (errors should be rare, and it does get freed when leaving the world and the arena is reset) and saves some extra code.

Reply:
I don't understand what is a problem with current implementation, could you elaborate?

Reply:
I want you to all following dupes and frees on these two variables.